pkg/crc/machine/vsock.go

```diff
@@ -48,25 +49,36 @@ const (
 	httpPort         = 80
 	httpsPort        = 443
 	apiPort          = 6443
+	cockpitPort      = 9090
 )
 
-func vsockPorts() []types.ExposeRequest {
-	return []types.ExposeRequest{
+func vsockPorts(ps preset.Preset) []types.ExposeRequest {
+	exposeRequest := []types.ExposeRequest{
 		{
 			Local:  fmt.Sprintf(":%d", constants.VsockSSHPort),
 			Remote: fmt.Sprintf("%s:%d", virtualMachineIP, internalSSHPort),
 		},
-		{
-			Local:  fmt.Sprintf(":%d", apiPort),
-			Remote: fmt.Sprintf("%s:%d", virtualMachineIP, apiPort),
-		},
-		{
-			Local:  fmt.Sprintf(":%d", httpsPort),
-			Remote: fmt.Sprintf("%s:%d", virtualMachineIP, httpsPort),
-		},
-		{
-			Local:  fmt.Sprintf(":%d", httpPort),
-			Remote: fmt.Sprintf("%s:%d", virtualMachineIP, httpPort),
-		},
 	}
+	if ps == preset.OpenShift {
+		exposeRequest = append(exposeRequest, []types.ExposeRequest{
+			{
+				Local:  fmt.Sprintf(":%d", apiPort),
+				Remote: fmt.Sprintf("%s:%d", virtualMachineIP, apiPort),
+			},
+			{
+				Local:  fmt.Sprintf(":%d", httpsPort),
+				Remote: fmt.Sprintf("%s:%d", virtualMachineIP, httpsPort),
+			},
+			{
+				Local:  fmt.Sprintf(":%d", httpPort),
+				Remote: fmt.Sprintf("%s:%d", virtualMachineIP, httpPort),
+			},
+		}...)
+	} else {
```

Comment:
please make the if checked named too and error when preset is not matched.
use the same pattern in the client.